docs/section21/Rule21-1.md

```diff
@@ -0,0 +1,45 @@
+
+# Boiler - Rule 21-1  
+
+**Rule ID:** 21-1  
+**Rule Description:** For systems using purchased hot water or steam, the heating source shall be modeled as purchased hot water or steam in both the proposed design and baseline building design. Hot-water or steam costs shall be based on actual utility rates.  
+**Rule Assertion:** B-RMR ASHRAE229:conditioning_component(external_fluid_source).type = P-RMR ASHRAE229:conditioning_component(external_fluid_source).type  
+**Appendix G Section:** Section 21 Boiler  
+**Appendix G Section Reference:** Section G3.1.1.1 & G3.1.1.3.1 Building System-Specific Modeling Requirements for the Baseline model  
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:**  
+
+1. P-RMR is modeled with purchased hot water or steam
+
+**Manual Check:** None  
+**Evaluation Context:** Building  
+**Data Lookup:** None  
+**Function Call:** None  
+
+## Rule Logic:  
+
+- For each conditioning component in P_RMR: `for conditioning_component_p in P_RMR.ASHRAE229.conditioning_components:`
+
+  - Check if conditioning component in P_RMR is purchased hot water or steam: `if ( conditioning_component_p IS ExternalFluidSource ) AND ( conditioning_component_p.type in ["HOT_WATER", "STEAM] ):`
+
+    - Set applicability flag: `rule_applicability_check = TRUE`
+
+    - Get matching external fluid source from B_RMR: `external_fluid_source_b = match_data_element(B_RMR, ExternalFluidSources, conditioning_component_p.id)`
+
```

Comment:
1. ServiceWaterHeatingSystem RMR is still in flux, be we may potentially have externalFluidSource apply to it as well. In that case, this RDS would need to differentiate between space heating and SWH.
2. App G is vague regarding handling of projects that have a mixture of purchased heat and on-site plants in the proposed design. So I am not sure we need to match data elements here. One potential interpretation is that if some systems in the proposed desing use purchased steam/hot water, all baseline systems must use purchased steam / hot water. For example, if proposed design has space heating via purchased steam + DOAS that uses on-site furnace, the baseline may be modeled with 100% purchased heat.

Reply:
@mkarpman i believe SWH is now approved and i dont believe there are any connects between SHW and externalfluidsource. Regarding (2) we can move all non-mandatory RDS to a separate branch and not discuss them right now. @mcneillj is that a good solution?